x-pack/plugins/cases/public/components/all_cases/table_filters.tsx

```diff
@@ -0,0 +1,201 @@
+}
+
+// Fix the width of the status dropdown to prevent hiding long text items
+const StatusFilterWrapper = styled(EuiFlexItem)`
+  && {
+    flex-basis: 180px;
+  }
+`;
+
+/**
+ * Collection of filters for filtering data within the CasesTable. Contains search bar,
+ * and tag selection
+ *
+ * @param onFilterChanged change listener to be notified on filter changes
+ */
+
+const defaultInitial = {
+  search: '',
+  reporters: [],
+  status: StatusAll,
+  tags: [],
+};
+
+const CasesTableFiltersComponent = ({
+  countClosedCases,
+  countOpenCases,
+  countInProgressCases,
+  onFilterChanged,
+  initial = defaultInitial,
+  setFilterRefetch,
+  disabledStatuses,
+}: CasesTableFiltersProps) => {
+  const [selectedReporters, setSelectedReporters] = useState(
+    initial.reporters.map((r) => r.full_name ?? r.username ?? '')
+  );
+  const [search, setSearch] = useState(initial.search);
+  const [selectedTags, setSelectedTags] = useState(initial.tags);
+  const { tags, fetchTags } = useGetTags();
+  const { reporters, respReporters, fetchReporters } = useGetReporters();
+
+  const refetch = useCallback(() => {
+    fetchTags();
+    fetchReporters();
+  }, [fetchReporters, fetchTags]);
+
+  useEffect(() => {
+    if (setFilterRefetch != null) {
+      setFilterRefetch(refetch);
+    }
+  }, [refetch, setFilterRefetch]);
+
+  useEffect(() => {
+    if (selectedReporters.length) {
+      const newReporters = selectedReporters.filter((r) => reporters.includes(r));
+      handleSelectedReporters(newReporters);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [reporters]);
+
+  useEffect(() => {
+    if (selectedTags.length) {
+      const newTags = selectedTags.filter((t) => tags.includes(t));
+      handleSelectedTags(newTags);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [tags]);
+
+  const handleSelectedReporters = useCallback(
+    (newReporters) => {
+      if (!isEqual(newReporters, selectedReporters)) {
+        setSelectedReporters(newReporters);
+        const reportersObj = respReporters.filter(
+          (r) => newReporters.includes(r.username) || newReporters.includes(r.full_name)
+        );
+        onFilterChanged({ reporters: reportersObj });
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [selectedReporters, respReporters]
+  );
+
+  const handleSelectedTags = useCallback(
+    (newTags) => {
+      if (!isEqual(newTags, selectedTags)) {
+        setSelectedTags(newTags);
+        onFilterChanged({ tags: newTags });
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [selectedTags]
+  );
+
+  const handleOnSearch = useCallback(
+    (newSearch) => {
+      const trimSearch = newSearch.trim();
+      if (!isEqual(trimSearch, search)) {
+        setSearch(trimSearch);
+        onFilterChanged({ search: trimSearch });
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
definitely test it but i think `react-hooks/exhaustive-deps` should be removed from the `useCallback`s in this file

Reply:
Yea, I've removed it. Yea, I was planning on removing all the `exhaustive-deps` ideally in cases in a follow up refactor